src/JwksClient.js

```diff
@@ -79,19 +79,20 @@ export class JwksClient {
             return ((key.x5c && key.x5c.length) || (key.n && key.e));
         })
         .map(key => {
-          if (key.x5c && key.x5c.length) {
-            return {
-              kid: key.kid,
-              nbf: key.nbf,
-              publicKey: certToPEM(key.x5c[0])
-            };
+          const hasCertificateChain = key.x5c && key.x5c.length;
+          const pem = hasCertificateChain ? certToPEM(key.x5c[0]) : rsaPublicKeyToPEM(key.n, key.e);
+          const jwk = {
+            kid: key.kid,
+            nbf: key.nbf,
+            pem
+          };
+          //below is for backwards compatibility
+          if (hasCertificateChain) {
```

Comment:
I think having the test here and having it again up above for setting pem makes this refactor more confusing than the original if/else blocks.

Reply:
I can roll this back